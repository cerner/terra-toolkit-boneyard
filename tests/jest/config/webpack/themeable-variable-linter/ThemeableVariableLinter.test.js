const ThemeableVariableLinter = require('../../../../../config/webpack/themeable-variable-linter/ThemeableVariableLinter');

describe('Themeable Variable Linter', () => {
  it('creates the tracker with a filled out config', () => {
    const themeConfig = {
      theme: 'default',
      scoped: ['a', 'b'],
    };
    const plugin = new ThemeableVariableLinter(themeConfig);
    expect(plugin.variableInformation).toMatchSnapshot();
  });

  it('creates the tracker with no default in the config', () => {
    const themeConfig = {
      scoped: ['a', 'b'],
    };
    const plugin = new ThemeableVariableLinter(themeConfig);
    expect(plugin.variableInformation).toMatchSnapshot();
  });

  it('creates the tracker with no scoped in the config', () => {
    const themeConfig = {
      theme: 'default',
    };
    const plugin = new ThemeableVariableLinter(themeConfig);
    expect(plugin.variableInformation).toMatchSnapshot();
  });

  it('creates the tracker with an undefined scoped in the config', () => {
    const themeConfig = {
      theme: 'default',
      scoped: ['a', 'b', undefined],
    };
    const plugin = new ThemeableVariableLinter(themeConfig);
    expect(plugin.variableInformation).toMatchSnapshot();
  });
});
