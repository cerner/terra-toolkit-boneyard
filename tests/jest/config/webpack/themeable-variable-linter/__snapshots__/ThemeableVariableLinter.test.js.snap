// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Themeable Variable Linter creates the tracker with a filled out config 1`] = `undefined`;

exports[`Themeable Variable Linter creates the tracker with an undefined scoped in the config 1`] = `undefined`;

exports[`Themeable Variable Linter creates the tracker with no default in the config 1`] = `undefined`;

exports[`Themeable Variable Linter creates the tracker with no scoped in the config 1`] = `undefined`;
